plugins/modules/network_settings_intent.py

```diff
@@ -0,0 +1,1937 @@
+            .update({"Network Details": self.want.get("wantNetwork").get("settings")})
+        return
+
+    def get_diff_merged(self, config):
+        """
+        Update/Create Global Pool Reserve Pool and
+        Network in DNAC with fields provided in DNAC
+
+        Parameters:
+            config - Playbook details
+
+        Returns:
+            self
+        """
+
+        if config.get("GlobalPoolDetails") is not None:
+            self.update_global_pool(config)
+
+        if config.get("ReservePoolDetails") is not None:
+            self.update_reserve_pool(config)
+
+        if config.get("NetworkManagementDetails") is not None:
+            self.update_network(config)
+
+        return self
+
+    def delete_reserve_pool(self, name):
+        """
+        Delete Reserve Pool in DNAC with fields provided in playbook.
+
+        Parameters:
+            name - Reserve pool name
+
+        Returns:
+            self
+        """
+
+        reserve_pool_exists = self.have.get("reservePool").get("exists")
+        self.log("Reserved Ip Pool to be deleted " +
+                 str(self.want.get("wantReserve").get("name")))
+
+        if not reserve_pool_exists:
+            self.msg = "Reserved Ip Subpool Not Found"
+            self.status = "failed"
+            return self
+
+        _id = self.have.get("reservePool").get("id")
+        self.log("Reserve pool {0} id ".format(name) + str(_id))
+        response = self.dnac._exec(
+            family="network_settings",
+            function="release_reserve_ip_subpool",
+            params={"id": _id},
+        )
+        self.check_execution_response_status(response).check_return_status()
+        executionid = response.get("executionId")
+        result_reserve_pool = self.result.get("response")[1].get("reservePool")
+        result_reserve_pool.get("response").update({name: {}})
+        result_reserve_pool.get("response").get(name) \
+            .update({"Execution Id": executionid})
+        result_reserve_pool.get("msg") \
+            .update({name: "Ip subpool reservation released successfully"})
+        self.msg = "Reserve pool - {0} released successfully".format(name)
+        self.status = "success"
+        return self
+
+    def delete_global_pool(self, name):
+        """
+        Delete Global Pool in DNAC with fields provided in playbook.
+
+        Parameters:
+            name - Global pool name
+
+        Returns:
+            self
+        """
+
+        global_pool_exists = self.have.get("globalPool").get("exists")
+        if not global_pool_exists:
+            self.msg = "Global pool Not Found"
+            self.status = "failed"
+            return self
+
+        response = self.dnac._exec(
+            family="network_settings",
+            function="delete_global_ip_pool",
+            params={"id": self.have.get("globalPool").get("id")},
+        )
+        self.check_execution_response_status(response).check_return_status()
```

Comment:
You can add comment.. it will be clear to read if we make blocks. For example,

```
    # Check the execution status
    self.check_execution_response_status(response).check_return_status()
    execution_id = response.get("executionId")

    # Update result information
    result_global_pool = self.result.get("response")[0].get("globalPool")
    result_global_pool.get("response").update({name: {}})
    result_global_pool.get("response").get(name).update({"Execution Id": execution_id})
    result_global_pool.get("msg").update({name: "Pool deleted successfully"})

    self.msg = "Global pool - {0} deleted successfully".format(name)
    self.status = "success"
```